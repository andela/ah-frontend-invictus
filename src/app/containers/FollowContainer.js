import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import followUser from '../../redux/actions/FollowingAction';
import FollowComponent from '../components/FollowComponent';


export class FollowContainer extends Component {
  componentDidMount () {
    const username = localStorage.getItem('username');
    const { followUser } = this.props;
    followUser(username);
  }


  render () {
    const { profiles } = this.props;
    return (
      <FollowComponent state={this.props} profiles={profiles} />
    );
  }
}

export const mapStateToProps = state => ({
  profiles: state.followUser,
  currentUser: state.auth.login_success.username,
  following: state.followUser
});

FollowContainer.propTypes = {
  followUser: PropTypes.func
};

export default connect(mapStateToProps, { followUser })(FollowContainer);
