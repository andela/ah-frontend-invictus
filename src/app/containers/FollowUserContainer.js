import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import followUsers from '../../redux/actions/followAction';
import FetchUsers from '../components/FetchUsers';


export class FollowUserContainer extends Component {
  followAGivenUser = (username) => {
    const { followUsers } = this.props;
    followUsers(username);
  }


  render () {
    const { profiles } = this.props;
    return (
      <FetchUsers state={this.props} profiles={profiles} followAGivenUser={this.followAGivenUser} />
    );
  }
}

export const mapStateToProps = state => ({
  profiles: state.followUsers,
  currentUser: state.auth.login_success.username,
  following: state.followUsers
});

FollowUserContainer.propTypes = {
  followUser: PropTypes.func,
  followAGivenUser: PropTypes.func
};

export default connect(mapStateToProps, { followUsers })(FollowUserContainer);
