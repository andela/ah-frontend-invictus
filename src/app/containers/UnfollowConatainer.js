import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import unfollowUsers from '../../redux/actions/UnfollowAction';
import FetchUsers from '../components/FetchUsers';


export class UnFollowContainer extends Component {
  unfollowAGivenUser = (username) => {
    const { unfollowUsers } = this.props;
    unfollowUsers(username);
  }


  render () {
    const { profiles } = this.props;
    return (
      <FetchUsers state={this.props} profiles={profiles} unfollowAGivenUser={this.unfollowAGivenUser} />
    );
  }
}

const mapStateToProps = state => ({
  profiles: state.unfollowUsers,
  currentUser: state.auth.login_success.username,
  unfollowing: state.unfollowUsers
});

UnFollowContainer.propTypes = {
  unfollowUser: PropTypes.func
};

export default connect(mapStateToProps, { unfollowUsers })(UnFollowContainer);
