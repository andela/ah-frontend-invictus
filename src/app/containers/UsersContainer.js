import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import fetchUser from '../../redux/actions/FetchUsersAction';
import followUser from '../../redux/actions/FollowingAction';
import followUsers from '../../redux/actions/followAction';
import FetchUsers from '../components/FetchUsers';
import unfollowUsers from '../../redux/actions/UnfollowAction';


export class UsersContainer extends Component {
  componentWillMount () {
    this.props.fetchUser();
    const username = localStorage.getItem('username');
    const { followUser } = this.props;
    followUser(username);
  }

  followAGivenUser = (username) => {
    const { followUsers } = this.props;
    followUsers(username);
  }

  unfollowAGivenUser = (username) => {
    const { unfollowUsers } = this.props;
    unfollowUsers(username);
  }
  render () {
    const { profiles } = this.props;
    return (
      <FetchUsers state={this.props} profiles={profiles}/>
    );
  }
}

export const mapStateToProps = state => ({
  profiles: state.fetchUser,
  currentUser: state.auth.login_success.username,
  following: state.followUser,
  followings: state.followUsers,
  unfollowing: state.unfollowUsers
});

UsersContainer.propTypes = {
  fetchUser: PropTypes.func,
  followAGivenUser: PropTypes.func,
  unfollowAGivenUser: PropTypes.func
};

export default connect(mapStateToProps, { fetchUser, followUser, followUsers, unfollowUsers })(UsersContainer);
