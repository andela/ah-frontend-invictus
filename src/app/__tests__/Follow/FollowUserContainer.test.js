import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { FollowUserContainer } from  '../../containers/FollowUserContainer';

describe('Testing FollowUserContainer', () => {
    const props = {
        followUsers: jest.fn(),
        followAGivenUser: jest.fn(),
    }

    it('matches snapshot', () => {
        const wrapper = shallow(<FollowUserContainer {...props } />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('it should call followAGivenUser and followUsers ', () => {
        const props = {
            followAGivenUser: jest.fn(),
            followUsers: jest.fn(),
        }
        const wrapper = shallow(<FollowUserContainer {...props } />);
        const instance = wrapper.instance();
        instance.followAGivenUser('username');
        expect(props.followUsers).toHaveBeenCalledWith('username');
        expect(props.followUsers).toHaveBeenCalledWith('username');
    });

});