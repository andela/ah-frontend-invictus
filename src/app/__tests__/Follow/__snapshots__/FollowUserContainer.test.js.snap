// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FollowUserContainer matches snapshot 1`] = `
<FetchUsers
  followAGivenUser={[Function]}
  state={
    Object {
      "followAGivenUser": [MockFunction],
      "followUsers": [MockFunction],
    }
  }
/>
`;
